frontrunner_sdk/commands/injective/get_transaction.py

```diff
@@ -0,0 +1,69 @@
+import json
+
+from collections import Iterable
+from dataclasses import dataclass
+from dataclasses import field
+from typing import List
+
+from pyinjective.proto.cosmos.tx.v1beta1.service_pb2 import GetTxResponse
+
+from frontrunner_sdk.commands.base import FrontrunnerOperation
+from frontrunner_sdk.ioc import FrontrunnerIoC
+from frontrunner_sdk.logging.log_operation import log_operation
+
+
+@dataclass
+class OrderFailure:
+  flags: List[int]
+  hashes: List[str]
+
+  EVENT_ORDER_FAIL_TYPE = "injective.exchange.v1beta1.EventOrderFail"
+
+  @classmethod
+  def from_injective_response(clz, injective_response: GetTxResponse) -> list:
+    result = []
+    if getattr(injective_response, "tx_response"):
+      tx_response = injective_response.tx_response
+      if getattr(tx_response, "logs"):
+        logs = tx_response.logs
+        if logs and issubclass(type(logs), Iterable):
+          for log in logs:
+            if getattr(log, "events") and issubclass(type(log.events), Iterable):
+              for event in log.events:
+                if event.type == clz.EVENT_ORDER_FAIL_TYPE:
+                  attributes = event.attributes
+                  flags: List[int] = next(
+                    (json.loads(attribute.value) for attribute in attributes if attribute.key == "flags"), []
+                  )
+                  hashes: List[str] = next(
+                    (json.loads(attribute.value) for attribute in attributes if attribute.key == "hashes"), []
+                  )
+                  result.append(clz(flags, hashes))
+
+    return result
```

Comment:
Can we do a guard clause style on this?

Also, are all of those entries possibly not there? IIRC, protobuf defaults them to sensible values, so even if you do `stuff.events` but events doesn't exist, it'll be None instead of an error.

Reply:
This is meant to guard against protobuf structural changes and prevent this:
```
a = injective_response.field_name
AttributeError: field_name
```

What guard clause style are you suggesting though?

Reply:
```python
if not hasattr(injective_response, "tx_response"):
  return []

tx_response = injective_respones.tx_response

if not hasattr(tx_response, "logs"):
  return []

try:
  logs = list(tx_response.logs)
except:
  return []

try:
  events = list(itertools.chain.from_iterable(log.events for log in logs))
except:
  return []

return [
  clz(
    next((json.loads(attribute.value) for attribute in attributes if attribute.key == "flags"), []),
    next((json.loads(attribute.value) for attribute in attributes if attribute.key == "hashes"), []),    
  )
  for event
  in events
  if event.type == clz.EVENT_ORDER_FAIL_TYPE
]
```

Reply:
Nice, updated